python_modules/dagster/dagster/core/definitions/graph_definition.py

```diff
@@ -465,8 +465,12 @@ def to_job(
         version_strategy: Optional[VersionStrategy] = None,
         op_selection: Optional[List[str]] = None,
         partitions_def: Optional["PartitionsDefinition"] = None,
-        asset_keys_by_input_handle: Optional[Mapping[SolidInputHandle, AssetKey]] = None,
-        asset_keys_by_output_handle: Optional[Mapping[SolidOutputHandle, AssetKey]] = None,
+        asset_keys_by_input_handle: Optional[
+            Mapping[Tuple[NodeHandle, InputDefinition], AssetKey]
```

Comment:
Thoughts on making a `NodeInputHandle` / `NodeOutputHandle`?

Reply:
+1. Would also be nice to add a comment here just to clarify that `NodeInputHandle`/`NodeOutputHandle` are unique within a graph